Mathlib/Analysis/LegendrePoly.lean

```diff
@@ -0,0 +1,140 @@
+/-
+Copyright (c) 2024 Junqi Liu. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Junqi Liu
+-/
+import Mathlib.Algebra.Polynomial.Basic
+import Mathlib.Algebra.BigOperators.Ring
+import Mathlib.Algebra.Order.BigOperators.Group.Finset
+import Mathlib.Algebra.Polynomial.AlgebraMap
+import Mathlib.Algebra.Polynomial.Derivative
+import Mathlib.Analysis.Normed.Module.Basic
+import Mathlib.Data.Nat.Choose.Sum
+import Mathlib.Data.Real.Basic
+import Mathlib.Tactic.Linarith
+import Mathlib.Tactic.Ring
+
+/-!
+# Legendre Polynomials
+
+In this file, we define the Legendre polynomials `Legendre n` for `n : ℕ` as a polynomial in `ℝ[X]`.
+We prove some basic properties of the Legendre polynomials.
+
+-/
+
+open scoped Nat
+open BigOperators Finset
+
+variable {R : Type*}
+
+namespace Polynomial
+
+/--  `Legendre n` is the polynomial defined in terms of derivatives of order n.  -/
+noncomputable def Legendre (n : ℕ) : ℝ[X] :=
+  C (n ! : ℝ)⁻¹ * derivative^[n] (X ^ n * (1 - X) ^ n)
+
+lemma Finsum_iterate_deriv [CommRing R] {k : ℕ} {h : ℕ → ℕ} :
```

Comment:
Here `k` and `h` should be explicit. Also, this lemma can go to a more basic file.

Reply:
ok! ok!